demisto_sdk/commands/common/content/objects/pack_objects/pack.py

```diff
@@ -64,7 +68,14 @@ def _content_files_list_generator_factory(self, dir_name: str, suffix: str) -> I
         """
         objects_path = (self._path / dir_name).glob(patterns=[f"*.{suffix}", f"*/*.{suffix}"])
         for object_path in objects_path:
-            yield path_to_pack_object(object_path)
+            content_object = path_to_pack_object(object_path)
+            # skip content items that are not displayed in the id set, if the corresponding flag is used
+            if self._filter_items_by_id_set and content_object.type().value not in [FileType.RELEASE_NOTES.value,
+                                                                                    FileType.RELEASE_NOTES_CONFIG.value]:
+                if is_object_in_id_set(content_object.get('name'), self._pack_info_from_id_set):
```

Comment:
let's use the file path to compare instead of the name so the file will not be opened

Reply:
As discussed, we'll leave it as is, since searching for the path causes iterating over all of the id set items of a certain type, since the items are ordered as a list of dicts (as opposed to the way it is currently, where the check is by `Packs` key in the id set, which is a single dict and can be addressed by the pack's name and `ContentItems` key)